### ISSUE TITLE
feature: Nix package or nix support?

### ISSUE THREAD

**Issue by Cyberskeptic** [kRHYME7/Hyde-cli#66]
### Did you check the docs?

- [X] I have read all the Hyde docs

### Is your feature request related to a problem? Please describe.

Installation doesnt work on nix, gives a make error:
install: target '/usr/lib/hyde-cli': No such file or directory
make: *** [Makefile:61: install] Error 1

### Describe the solution you'd like

Either a nix package or an install script that supports nix.

### Describe alternatives you've considered

I might be able to maybe manually install and move all the files and hunt down all the packages.

**Comment by kRHYME7** [comment]
> install: target '/usr/lib/hyde-cli': No such file or directory

Hi this CLI cannot be used in nix as this is designed to be used in `/usr/` ` /etc `  so only the typical linux package manager can be use for this. For nix we probably need to port or add some conditions for nix. 

I think we can populate this inside `  ~/.local/ ` which is under the $USER scope? 

I am not using nix, so can you provide me the file paths that are equivalent on nix?  

* `/usr/share/hyde-cli `
* ` /etc/hyde-cli ` 
* ` /usr/bin `

**Comment by Cyberskeptic** [comment]
The `~/.local/` should work

**Comment by kRHYME7** [comment]
my "guess" is /etc has the ` ~/.config ` counterpart right?  share and bin will in in local. 

so we can make 

```
~/.local/lib/hyde-cli
~/.local/share/hyde-cli
~/.local/bin/hyde-cli
```
But again we need to make a good port of HyDE. Can we do it on home Manager?

**Comment by Cyberskeptic** [comment]
Not the most experienced nix user but I think this is a home manager type thing.

**Comment by kRHYME7** [comment]
https://github.com/kRHYME7/Hyde-cli?tab=readme-ov-file#installation


you can install the CLI locally now, but this is just for future proofing as we do not have installer for another distro

**Comment by Cyberskeptic** [comment]
That didnt work so I might sit down and make a nix package at some point.

**Comment by Cyberskeptic** [comment]
There was no error, the operation completes but the commands arent there basically no matter how I try using it, and yes I have make installed so it is using the make file, the output is normal.

**Comment by kRHYME7** [comment]
Cool you needed to add  ` ~/.local/bin ` to PATH or how nixOS handles  this ? 

![image](https://github.com/kRHYME7/Hyde-cli/assets/53417443/a28b8aa9-b8c9-49be-885f-f52db5a0dc76)

![image](https://github.com/kRHYME7/Hyde-cli/assets/53417443/242cf7dd-f4e8-4be3-a313-03652d660cd4)

**Comment by sandeng1440** [comment]
> Cool you needed to add `~/.local/bin` to PATH or how nixOS handles this ?
> 

This is true. Basically, nix works like all other distros. You don't need to use most nix features to install HyDE. In fact, you must avoid coupling it with nix features like home-manager if you want to be able to change themes on the fly without having to rebuild the system or home-manager.
